main/influxdb_watcher.go

```diff
@@ -0,0 +1,209 @@
+	}
+	message := Message{
+		UUID:     res[0].Series[0].Values[0][2].(string),
+		SentTime: res[0].Series[0].Values[0][1].(string),
+	}
+	bytes, err := json.Marshal(message)
+	if err != nil {
+		panic("json encoding failed")
+	}
+	return &bytes, nil
+}
+
+// queryDB convenience function to query the database
+func queryDB(broker *InfluxdbBroker, cmd string) (res []client.Result, err error) {
+	q := client.Query{
+		Command:  cmd,
+		Database: broker.MonascaDB,
+	}
+	if response, err := broker.Connection.Query(q); err == nil {
+		if response.Error() != nil {
+			return res, response.Error()
+		}
+		res = response.Results
+	} else {
+		return res, err
+	}
+	return res, nil
+}
+
+func main() {
+	configuration := watcherConfiguration{}
+	err := configEnv.Parse(&configuration)
+	if err != nil {
+		log.Fatalf("Failed to parse environment variables. %s", err)
+	}
+
+	influxdbAddress := configuration.InfluxdbAddress
+	validateConfigurationString("INFLUXDB_ADDRESS", influxdbAddress)
+	username := configuration.Username
+	validateConfigurationString("INFLUXDB_USERNAME", username)
+	password := configuration.Password
+	validateConfigurationString("INFLUXDB_PASSWORD", password)
+	prometheusEndpoint := configuration.PrometheusEndpoint
+	validateConfigurationString("PROMETHEUS_ENDPOINT", prometheusEndpoint)
+
+	log.Infof("Using InfluxDB Address %s", influxdbAddress)
+
+	influxdbBroker := InfluxdbBroker{MonascaDB: "mon"}
+	period := time.Duration(configuration.Period) * time.Second
+	timeout := time.Duration(configuration.Timeout) * time.Second
+
+	log.Infof(period.String())
+
+	// Create the watcher now so it has NOT_STARTED status while we are initializing
+	// the connections to InfluxDB
+	watcher := watcher.CreateWatcher(&influxdbBroker, period, 1, timeout, "influxdb")
+
+	log.Infof(watcher.Period.String())
+
+	go func() {
+		// Start prometheus endpoint
+		http.Handle("/metrics", promhttp.Handler())
+		log.Fatal(http.ListenAndServe(prometheusEndpoint, nil))
+	}()
+
+	var connection client.Client
+	connection, err = startConnect(influxdbAddress, username, password)
+	if err != nil {
+		log.Fatalf("Starting connection to InfluxDB failed: %s", err)
+	}
+	influxdbBroker.Connection = connection
+	watcher.Start()
+
+	log.Info("Started influxdb-watcher")
+
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	sigchan := make(chan os.Signal)
+	signal.Notify(sigchan, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		defer wg.Done()
+		defer watcher.Stop()
+		sig := <-sigchan
+
+		log.Fatalf("Caught signal %v: terminating", sig)
+	}()
+
+	log.Infof("Serving metrics on %s/metrics", prometheusEndpoint)
+	wg.Wait()
+}
+
+func startConnect(influxdbAddress string, username string, password string) (client.Client, error) {
+	c, err := client.NewHTTPClient(client.HTTPConfig{
+		Addr:     influxdbAddress,
+		Username: username,
+		Password: password,
+	})
+	if err != nil {
+		return nil, err
+	}
+	return c, nil
+}
+
+func validateConfigurationString(name, value string) {
```

Comment:
1. Would prefer shorter name like `validateConfString` or even `validateStr`
2. I  think funcs like this should be moved to `github.com/monasca/monasca-watchers/watcher` or some common package.